### PR TITLE
Revert "TableOfContent now accepts Handle, not Runtime (#2096)"

### PR DESCRIPTION
**Pull request by agourlay**
This PR unfortunately reverts the great work done in https://github.com/qdrant/qdrant/pull/2096

This is necessary because shutting down an instance with collections currently yields the following logs:

```
[2023-07-03T09:44:12.144Z WARN  collection::shards::local_shard] Update workers failed with: Service internal error: task 395 was cancelled
[2023-07-03T09:44:12.148Z INFO  wal] segment creator shutting down
[2023-07-03T09:44:12.149Z WARN  collection::shards::local_shard] Error sending stop signal to update handler: channel closed
[2023-07-03T09:44:12.149Z WARN  collection::update_handler] Failed to stop flush worker as it is already stopped.
[2023-07-03T09:44:12.149Z WARN  collection::shards::local_shard] Update workers failed with: Service internal error: task 123 was cancelled
[2023-07-03T09:44:12.152Z INFO  wal] segment creator shutting down
[2023-07-03T09:44:12.153Z WARN  collection::shards::local_shard] Error sending stop signal to update handler: channel closed
[2023-07-03T09:44:12.153Z WARN  collection::update_handler] Failed to stop flush worker as it is already stopped.
[2023-07-03T09:44:12.153Z WARN  collection::shards::local_shard] Update workers failed with: Service internal error: task 127 was cancelled
[2023-07-03T09:44:12.157Z INFO  wal] segment creator shutting down
[2023-07-03T09:44:12.159Z WARN  collection::shards::local_shard] Error sending stop signal to update handler: channel closed
[2023-07-03T09:44:12.159Z WARN  collection::update_handler] Failed to stop flush worker as it is already stopped.
[2023-07-03T09:44:12.159Z WARN  collection::shards::local_shard] Update workers failed with: Service internal error: task 231 was cancelled
[2023-07-03T09:44:12.162Z INFO  wal] segment creator shutting down
[2023-07-03T09:44:12.163Z WARN  collection::shards::local_shard] Error sending stop signal to update handler: channel closed
[2023-07-03T09:44:12.163Z WARN  collection::update_handler] Failed to stop flush worker as it is already stopped.
[2023-07-03T09:44:12.163Z WARN  collection::shards::local_shard] Update workers failed with: Service internal error: task 203 was cancelled
[2023-07-03T09:44:12.168Z INFO  wal] segment creator shutting down
```

The system cannot shutdown properly because a Tokio runtime is not available when dropping collections.

This happens because the refactoring of the Tokio handles was merged **after** the concurrent PR https://github.com/qdrant/qdrant/pull/2144 which requires to have access to a runtime when dropping a collection.

I thought an easy fix was to simply shutdown the `toc` before the runtimes but unfortunately it brings a different set of issues as discovered in https://github.com/qdrant/qdrant/pull/2190

The bottom line is that more work is required in order to have a clean shutdown without noisy logs in the presence of telemetry and/or consensus.

Given that we need to cut a release very soon, the most sensible approach is to revert and research how to re-integrate the change afterwards without pressure.

I apologize for the bad turn of events @bazhenov as you are not responsible for this integration issue :bow:

### CODE REVIEW

**Review by generall — APPROVED** [review]

**Review by timvisee — APPROVED** [review]

File: src/consensus.rs [hunk 1]
```diff
@@ -94,22 +86,20 @@ impl Consensus {
...
+        thread::Builder::new()
+            .name("consensus".to_string())
+            .spawn(move || {
+                if let Err(err) = consensus.start() {
+                    log::error!("Consensus stopped with error: {err}");
+                    state_ref_clone.on_consensus_thread_err(err);
+                } else {
+                    log::info!("Consensus stopped");
+                    state_ref_clone.on_consensus_stopped();
+                    state_ref_clone.on_consensus_stopped();
```
Comment (timvisee) [inline src/consensus.rs:100]: We should probably remove this line, because it is a duplicate.

Comment (timvisee) [reply src/consensus.rs:100]: Was just a second too late. I'll open a separate PR for this right now.

Comment (timvisee) [reply src/consensus.rs:100]: https://github.com/qdrant/qdrant/pull/2192

Comment (generall) [reply src/consensus.rs:100]: will submit as another PR

**Comment by bazhenov** [comment]
Thank you for the mention. Those kind of integrations issues are inevitable in real projects so I'm totally ok with that.

I can investigate it further if you don't mind.

**Comment by agourlay** [comment]
@bazhenov happy to hear you are still interested in investigating this :+1:
